frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -104,6 +108,12 @@ public final class EnhancedGraphTable extends BaseEnhancedGraphTable
   @UIEffect
   private @Nullable String selectedBranchName;
 
+  // To accurately track the change of the current branch from the begging, let's put something impossible
+  // as a branch name. This is required to detect the moment when the unmanaged branch notification should be shown.
+  private @Nullable String currentBranch = "?!@#$%^&";
+  private @Nullable Notification slideInNotification;
+  private @Nullable IGitMacheteRepository gitMacheteRepository;
```

Comment:
TBD for both mutable fields if they can be `@UIEffect`, which would save us the hassle with race conditions